### PR TITLE
Fail tests when they leave garbage behind

### PR DESCRIPTION
**Pull request by egonelbre**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -84,7 +84,7 @@ all-in-one: ## Deploy docker images with one storagenode locally
 		&& export VERSION="${TAG}"; \
 	fi \
 	&& docker-compose up -d storagenode \
-	&& scripts/fix-mock-overlay \
+	&& scripts/fix-mock-overlay.sh \
```
Comment (brimstone) [inline Makefile]: It's a bash script though, not an sh script. I'm of the mind that if the script has an executable bit set, it changes the environment and the system knows how to run it. The extensions are for humans to know which interpreter or compiler to use.

Comment (egonelbre) [reply Makefile]: Hmm, I need a nice way to distinguish allowed and non-allowed files. The goal is to fail tests when they create garbage into source directories.

Hmm, I could check the first few bytes to check whether they have a shebang?

Comment (brimstone) [reply Makefile]: or maybe if `git status` shows untracked files?

Comment (egonelbre) [reply Makefile]: We want it to fail even when files are in `.gitignore`. The goal is to ensure that development directory is left clean for all developers. `.gitignore` is for developers from accidentally committing files.

Comment (brimstone) [reply Makefile]: Then have travis delete its `.gitignore` then check for anything changed or untracked.

Comment (brimstone) [reply Makefile]: Also, is a test leaving a file that's `.gitignore`d a bad thing in this case? The developer isn't going to check it in, travis isn't going to find it so it'll just mess up the single developer's environment if anything.

Comment (egonelbre) [reply Makefile]: Oh nice, there's actually a command that lists untracked files ignoring .gitignore `git ls-files .  --others`.

Comment (egonelbre) [reply Makefile]: I'll close this and make a new PR.

Comment (brimstone) [reply Makefile]: Neat!

Comment (egonelbre) [reply Makefile]: https://github.com/storj/storj/pull/667

**Pull request closed without merging**